### ISSUE TITLE
Support --base-path

### ISSUE THREAD

**Issue by KPrasch** [iamdefinitelyahuman/py-solc-x#78]
### Overview
I am trying to leverage the `--base-path` solc feature introduced in version 0.6.9 to reduce complexity handling import remappings, however, there is no kwarg for `solc_wrapper` mapping to this flag.

### Specification

Ideally this feature can be implemented like other CLI flags as a keyword argument:
```python
solc_wrapper(base_path='/some/source/path/dir/', ...)
```

**Comment by iamdefinitelyahuman** [comment]
Yeah, that seems useful and easy to expose.  I'll try to get a new version out asap.

**Comment by iamdefinitelyahuman** [comment]
OK I added a quick fix so it's not a blocker.  But I also noticed there are quite a few recently added kwargs that lack support!

Long term I think the kwargs should be generalized / future-proofed by regexing the `solc --help` output.  This would also reduce a lot of overhead around checking exactly which flags were added in which version.  A few of them require specific logic but the fast majority are just setting a flag or setting a flag and adding a string after it.. shouldn't be that hard to do.

All that said - this will require a decent amount of refactoring, so it's a task for the next major release I think.
